src/tensors/levels/sparsehashlevels.jl

```diff
@@ -75,17 +75,17 @@ function moveto(lvl::SparseHashLevel{N, TI, Ptr, Tbl, Srt, Lvl}, device) where {
     return SparseHashLevel{N, TI, Ptr, Tbl, typeof(ptr_2), typeof(srt_2), typeof(lvl_2)}(lvl_2, lvl.shape, ptr_2, tbl_2, srt_2)
 end
 
-pattern!(lvl::SparseHashLevel{N, TI, Ptr, Tbl, Srt, Lvl}) where {N, TI, Ptr, Tbl, Srt, Lvl} = 
+pattern!(lvl::SparseHashLevel{N, TI, Ptr, Tbl, Srt, Lvl}) where {N, TI, Ptr, Tbl, Srt, Lvl} =
```

Comment:
Adding lots of whitespace comments makes it hard to see the relevant code that has changed.

Reply:
Yeah, I'll try to turn this off. I had to use a VSCode formatter for my last internship, and I haven't gotten rid of it. It pretty much just removes trailing spaces.

Reply:
Ah I see! No, you're good, let's remove trailing spaces! maybe I can figure out a way to do a PR that removes all of them once and for all.

Reply:
anyways, no worries for now, merge this thing!